From 64385bac7e1191eeebd2f2f930568adb3491dd9e Mon Sep 17 00:00:00 2001
From: hienhuynh <hien.huynh.px@rvc.renenas.com>
Date: Tue, 17 Apr 2018 09:56:36 +0700
Subject: [PATCH 080/504] ARM: dts r8a7742: iwg21m: Enable UHS SDR50 mode for
 sdhi2

This patch adds UHS-I pin state with power-source 1.8V to enable UHS-I
(SDH50) for SDHI2.

Signed-off-by: hienhuynh <hien.huynh.px@rvc.renenas.com>
Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
Signed-off-by: Loc Vu <loc.vu.zn@renesas.com>
---
 arch/arm/boot/dts/r8a7742-iwg21m.dts | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/r8a7742-iwg21m.dts b/arch/arm/boot/dts/r8a7742-iwg21m.dts
index 8073946..28ac7e8 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21m.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21m.dts
@@ -86,6 +86,13 @@
 	sdhi2_pins: sd2 {
 		groups = "sdhi2_data4", "sdhi2_ctrl";
 		function = "sdhi2";
+		power-source = <3300>;
+	};
+
+	sdhi2_pins_uhs: sd2_uhs {
+		groups = "sdhi2_data4", "sdhi2_ctrl";
+		function = "sdhi2";
+		power-source = <1800>;
 	};
 };
 
@@ -107,7 +114,8 @@
 
 &sdhi2 {
 	pinctrl-0 = <&sdhi2_pins>;
-	pinctrl-names = "default";
+	pinctrl-1 = <&sdhi2_pins_uhs>;
+	pinctrl-names = "default", "state_uhs";
 	vmmc-supply = <&vcc_sdhi2>;
 	vqmmc-supply = <&vccq_sdhi2>;
 
-- 
2.7.4

